core/tauri-utils/src/config.rs

```diff
@@ -565,6 +565,11 @@ pub struct MacConfig {
   /// Identity to use for code signing.
   #[serde(alias = "signing-identity")]
   pub signing_identity: Option<String>,
+  /// Whether the codesign should enable [hardened runtime] (for executables) or not.
+  ///
+  /// [hardened runtime]: <https://developer.apple.com/documentation/security/hardened_runtime>
+  #[serde(alias = "hardened-runtime", default = "hardened_runtime")]
```

Comment:
```suggestion
  #[serde(alias = "hardened-runtime", default = "default_true")]
```

Just re-use the default_true function from (defined at the top of the file)